dcm4che-imageio-opencv/src/main/java/org/dcm4che3/opencv/NativeJLSImageWriter.java

```diff
@@ -115,14 +115,15 @@ public void write(IIOMetadata streamMetadata, IIOImage image, ImageWriteParam pa
                 mat = ImageConversion.toMat(renderedImage, param.getSourceRegion(), false);
 
                 int jpeglsNLE = param instanceof JPEGLSImageWriteParam ? ((JPEGLSImageWriteParam) param).getNearLossless() : 0;
+                int bitCompressed = desc.getBitsCompressed();
                 int cvType = mat.type();
-                int elemSize = (int) mat.elemSize1();
                 int channels = CvType.channels(cvType);
-                int dcmFlags = CvType.depth(cvType) == CvType.CV_16S ? Imgcodecs.DICOM_IMREAD_SIGNED
-                    : Imgcodecs.DICOM_IMREAD_UNSIGNED;
-                if((dcmFlags & Imgcodecs.DICOM_IMREAD_SIGNED) == Imgcodecs.DICOM_IMREAD_SIGNED) {
+                boolean signed = desc.isSigned();
+                int dcmFlags = signed ? Imgcodecs.DICOM_FLAG_SIGNED : Imgcodecs.DICOM_FLAG_UNSIGNED;
+                if(signed) {
                     LOGGER.warn("Force compression to JPEG-LS lossless as lossy is not adapted to signed data.");
                     jpeglsNLE = 0;
+                    bitCompressed = 16; // Extend to bit allocated to avoid exception as negative values are treated as large positive values
```

Comment:
Is already handled by https://github.com/dcm4che/dcm4che/issues/638 - which allows to change the default behavior and restrict the compression to Bit Stored also for signed data, relying that receiving applications behaves according [DICOM PS 3.5, Pixel Data Encoding of Related Data Elements](dicom.nema.org/medical/dicom/current/output/html/part05.html#sect_8.1.1)
> Receiving applications may not assume anything about the contents of unused bits, and in particular may not assume that they are zero, or that they contain sign extension bits.